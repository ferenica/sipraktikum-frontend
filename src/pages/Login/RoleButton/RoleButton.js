import React from 'react';
import IconAccount from '../../../assets/GroupA/account.png';
import PropTypes from 'prop-types';

import {styled} from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';

const InactiveStyledCard = styled(Card)({
  'width': '310px',
  'display': 'flex',
  'alignItems': 'center',
  'justifyContent': 'center',
  'borderRadius': 10,
  'transition': '.15s all ease-in-out',
  '&:hover': {
    transform: 'scale(1.05)',
    backgroundColor: '#FFDEC5',
  },
  'margin': '3px',
});

const ActiveStyledCard = styled(InactiveStyledCard)({
  transform: 'scale(1.05)',
  backgroundColor: '#FFDEC5',
});

const StyledCardContent = styled(CardContent)({
  fontSize: '17px',
  fontWeight: 'bolder',
  lineHeight: '27px',
  whiteSpace: 'pre',
  alignItems: 'center',
  justifyContent: 'center',
  textAlign: 'center',
});

const StyledSSOText = styled('div')({
  fontSize: '14px',
  fontWeight: 'normal',
});

const RoleButton = (props) => {
  let ssoText;
  if (props.isSSO) {
    ssoText = (<StyledSSOText>SSO</StyledSSOText>);
  }

  if (props.isActive) {
    return (
      <ActiveStyledCard
        onClick={() => props.setType(props.name)}
        color="primary">
        <CardActionArea>
          <StyledCardContent>
            <img
              className="icon-account"
              src={IconAccount}
              style={{width: '20%'}}
              alt=""></img>
            <br></br><br></br>
            <div>{props.name}</div>
            {ssoText}
          </StyledCardContent>
        </CardActionArea>
      </ActiveStyledCard>
    );
  } else {
    return (
      <InactiveStyledCard
        onClick={() => props.setType(props.name)}
        color="primary">
        <CardActionArea>
          <StyledCardContent>
            <div className="content-role">
              <img
                className="icon-account"
                src={IconAccount}
                style={{width: '20%'}}
                alt=""></img>
              <br></br><br></br>
              <div>{props.name}</div>
              {ssoText}
            </div>
          </StyledCardContent>
        </CardActionArea>
      </InactiveStyledCard>
    );
  }
};

export default RoleButton;

RoleButton.propTypes = {
  name: PropTypes.string,
  setType: PropTypes.func,
  isActive: PropTypes.bool,
  isSSO: PropTypes.bool,
};
