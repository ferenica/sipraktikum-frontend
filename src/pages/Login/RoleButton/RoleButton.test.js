import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import RoleButton from './RoleButton';

Enzyme.configure({adapter: new EnzymeAdapter()});

describe('Halaman detail lembaga', () => {
  it('should renders correctly (inactive)', () => {
    const component = shallow(<RoleButton
      name='Supervisor Lembaga'
      isActive={false} />);
    expect(component).toMatchSnapshot();
  });

  it('should renders correctly (inactive)', () => {
    const component = shallow(<RoleButton
      name='Supervisor Lembaga'
      isActive={true} />);
    expect(component).toMatchSnapshot();
  });
});
