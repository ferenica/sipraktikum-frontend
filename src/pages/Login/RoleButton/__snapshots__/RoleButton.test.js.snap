// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Halaman detail lembaga should renders correctly (inactive) 1`] = `ShallowWrapper {}`;

exports[`Halaman detail lembaga should renders correctly (inactive) 2`] = `ShallowWrapper {}`;
